src/utils.ts

```diff
@@ -0,0 +1,46 @@
+import { Principal } from "@dfinity/principal";
+import { IndefiniteEvent, DetailValue } from "./declarations/cap/root";
+
+type DetailType =
+  | Principal
+  | bigint
+  | boolean
+  | string
+  | number
+  | Array<number>
+  | Array<DetailType>;
+type PrettyDetail = { [key: string]: DetailType };
+type DetailVector = { Vec: Array<DetailValue> };
+type GeneralValue = { [key: string]: any };
+
+const decodeDetailValue = (value: DetailValue): DetailType => {
+  const type = Object.keys(value)?.[0];
+  switch (type) {
+    case "Vec":
+      return (value as DetailVector)?.["Vec"]?.map(decodeDetailValue);
+    case "True":
+      return true;
+    case "False":
+      return false;
+    default:
+      return (value as GeneralValue)?.[type];
+  }
+};
+
+export const preattifyCapTransactions = (transaction: IndefiniteEvent) => {
+  const details = transaction?.details?.reduce<PrettyDetail>(
+    (acum, [key, value]) => ({
+      ...acum,
+      [key]: decodeDetailValue(value),
+    }),
+    {} as PrettyDetail
+  );
```

Comment:
This is missing the canister ID which is really needed. @b0xtch @qti3e does that come in the details or where would it be located?

Reply:
Also, with the canister Id we can aggregate DAB info

Reply:
The details, as expected for the UI are indeed correct. This is a utility method to parse the `details`, as @aleacevedo correctly handled (haven't tested but will put in use in cap-explorer to confirm and seems correct).